ubuntu_image/builder.py

```diff
@@ -403,53 +402,38 @@ def make_disk(self):
             image = Image(self.disk_img, self.image_size)
         offset_writes = []
         part_offsets = {}
-        root_offset = 0
         for i, part in enumerate(volume.structures):
             if part.name is not None:
                 part_offsets[part.name] = part.offset
             if part.offset_write is not None:
                 offset_writes.append((part.offset, part.offset_write))
-            image.copy_blob(self.boot_images[i],
+            image.copy_blob(self.part_images[i],
                             bs='1M', seek=part.offset // MiB(1),
                             count=ceil(part.size / MiB(1)),
                             conv='notrunc')
-            if part.type in ('bare', 'mbr'):
+            if (part.role is StructureRole.mbr or
+                    part.type == 'bare'):
                 continue
             # sgdisk takes either a sector or a KiB/MiB argument; assume
             # that the offset and size are always multiples of 1MiB.
             #
             # XXX Size must not be zero, which will happen if part.size < 1MiB
             partition_args = dict(
-                new='{}M:+{}M'.format(
-                    part.offset // MiB(1), part.size // MiB(1)),
+                new='{}M:+{}K'.format(
+                    part.offset // MiB(1), ceil(part.size / 1024)),
```

Comment:
If you look at the old code for the builder, you'll see that there were two image.partition() calls made - one for each partition and one for the rootfs, each of them having a different 'new' format ('{}M:+{}K' for rootfs, '{}M:+{}M' for normal). Now we just need one of them, so it was just a matter of deciding which of the two should stay. I decided on leaving the one that was used for the rootfs.

So in fact it's not that this 'changed', it's just about which of the two was removed. I can switch to the other one if you want though.